data/hlint.yaml

```diff
@@ -485,14 +484,18 @@
     - warn: {lhs: sequence_ (zipWith f x y), rhs: Control.Monad.zipWithM_ f x y}
     - warn: {lhs: sequence (replicate n x), rhs: Control.Monad.replicateM n x}
     - warn: {lhs: sequence_ (replicate n x), rhs: Control.Monad.replicateM_ n x}
+    - warn: {lhs: sequenceA (zipWith f x y), rhs: Control.Monad.zipWithM f x y}
+    - warn: {lhs: sequenceA_ (zipWith f x y), rhs: Control.Monad.zipWithM_ f x y}
+    - warn: {lhs: sequenceA (replicate n x), rhs: Control.Monad.replicateM n x}
+    - warn: {lhs: sequenceA_ (replicate n x), rhs: Control.Monad.replicateM_ n x}
     - warn: {lhs: mapM f (replicate n x), rhs: Control.Monad.replicateM n (f x)}
     - warn: {lhs: mapM_ f (replicate n x), rhs: Control.Monad.replicateM_ n (f x)}
     - warn: {lhs: mapM f (map g x), rhs: mapM (f . g) x, name: Fuse mapM/map}
     - warn: {lhs: mapM_ f (map g x), rhs: mapM_ (f . g) x, name: Fuse mapM_/map}
     - warn: {lhs: traverse f (map g x), rhs: traverse (f . g) x, name: Fuse traverse/map}
     - warn: {lhs: traverse_ f (map g x), rhs: traverse_ (f . g) x, name: Fuse traverse_/map}
-    - warn: {lhs: mapM id, rhs: sequence}
-    - warn: {lhs: mapM_ id, rhs: sequence_}
+    - warn: {lhs: mapM id, rhs: sequenceA}
+    - warn: {lhs: mapM_ id, rhs: sequenceA_}
```

Comment:
mapM has a Monad constraint, so does sequence, but sequenceA doesn't - so the original replacement seems right.

Reply:
That shouldn't be a problem since Applicative is less restrictive. The idea is to suggest the more general variant.

Reply:
At work we have rules like `- warn: {lhs: sequence, rhs: sequenceA}` to ban `sequence` (and also `mapM`, `forM`, `return` and so on). My guess is most people prefer their `Applicative` counterparts.

Reply:
I tend to reach for sequence still, since it's a shorter name, and `sequenceA` is visually ugly. I think it's probably good to have a group of rules (off-by-default) that do the `sequence => sequenceA`, that we might turn on by default one day, but not sure we want to be changing the type people use just yet (in contrast, if they use `traverse id`, that should definitely be `sequenceA`).

Currently I think return => pure is about as brave as I'm ready to be in HLint, and even then, I'm treading slowly (it's in the future group for now)

Reply:
Ok, makes sense! I've reverted those changes.